hathor/transaction/storage/block_height_index.py

```diff
@@ -48,15 +57,58 @@ def get(self, height: int) -> Optional[bytes]:
         """
         if len(self._index) <= height:
             return None
-        return self._index[height]
+        return self._index[height].hash
 
     def get_tip(self) -> bytes:
         """ Return the best block hash, it returns the genesis when there is no other block
         """
-        return self._index[-1]
+        return self._index[-1].hash
 
     def get_height_tip(self) -> Tuple[int, bytes]:
         """ Return the best block height and hash, it returns the genesis when there is no other block
         """
         height = len(self._index) - 1
-        return height, self._index[height]
+        return height, self._index[height].hash
+
+    def search_previous_best_block(self, timestamp: float) -> Optional[bytes]:
+        """Get the hash of the block with highest timestamp that isn't higher then the given timestamp."""
+        # XXX: commented simple O(N) search, use when in doubt
+        # for entry in reversed(self._index):
```

Comment:
Maybe implement this in a separate method so we can use in the tests.

Reply:
Moved to here: https://github.com/HathorNetwork/hathor-core/pull/295/files#r703133808